pkg/services/provisioning/dashboards/file_reader.go

```diff
@@ -51,34 +51,25 @@ func NewDashboardFileReader(cfg *DashboardsAsConfig, log log.Logger) (*fileReade
 	}, nil
 }
 
-func (fr *fileReader) ReadAndListen(ctx context.Context) error {
-	if err := fr.startWalkingDisk(); err != nil {
-		fr.log.Error("failed to search for dashboards", "error", err)
-	}
-
-	ticker := time.NewTicker(time.Duration(int64(time.Second) * fr.Cfg.UpdateIntervalSeconds))
-
-	running := false
-
+// pollChanges periodically runs startWalkingDisk based on interval specified in the config.
+func (fr *fileReader) pollChanges(ctx context.Context) {
+	ticker := time.Tick(time.Duration(int64(time.Second) * fr.Cfg.UpdateIntervalSeconds))
```

Comment:
Guess this changes behaviour: earlier the disk was walked straight away, but not it'll wait `UpdateIntervalSeconds` seconds before walking the disk?

Reply:
For this function yes, but it was only used in the `provision.go` and that is implemented in way that it still does the provisioning right away (either during startup or after during reload) and then kicks off the polling. I separated it so I could interweave some code in between and return from the API directly after the initial provisioning.